### DISCUSSION TITLE
Add 'select all' and 'invert selection' options to query results

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5624] [Feature Requests]
> When a user selects some records from the query list a "deselect all" button appears. However it would be very useful to also have options to "select all" and "invert selection" buttons that are always available

**Reported By:** **[CSIRO](https://app.asana.com/0/1201972579963460/1202844554923815/f)**

**Comment by maxpatiiuk** [comment]
Those seemingly simple features are actually quite hard to implement, as not all query results are loaded in the browser.

I.e, if query returns 10k rows, the front-end can't fetch all 10k because that would take a long time.

Also, it is extremely impractical to find specific few rows you want to unselect among the 10k results.

See https://github.com/specify/specify7/issues/1406 for a longer explanation.

Potential workarrounds:
1. Add "Select All" and "Invert Selection", but have them only appear when query has fewer than x result rows.
2. Do "Select All" and "Invert Selection" on the back-end (quite complicated)

Instead, most use cases for these features can be satisfied like this:
1. You can quickly select several rows by selecting first row, and then shift-clicking on the other row, and all rows in between would be selected.
2. You can adjust the query filters to exclude the rows you don't want to select.

**Comment by danb213** [comment]
Something that might help is to auto-tick the results that appear on first load of the query results, this would make it easier for users to either click "deselect all" to select a few results, or manually remove any outliers. In addition it would help clarify the process - after our UAT a common comment was on how to select all results, even though the system treats the list as being fully selected until at least one row is ticked. With the addition of a permanent "select all" and "deselect all" perhaps this behaviour needs to be changed?

I think it would be ok if the "select all" and "invert" only appeared for up to x rows for performance reasons, that should satisfy our users.

**Comment by maxpatiiuk** [comment]
From @carlosmbe:

> The[ Discourse Documentation ](https://discourse.specifysoftware.org/t/simple-search/185)states that simple search has a Select All/Deselect All Button. However, we only have a deselect button and the only way to select multiple items is manually. <img alt="Screenshot 2023-06-16 at 2 56 39 PM" width="783" src="https://user-images.githubusercontent.com/53784701/246522242-06a26f49-5d6c-4e60-bc2e-3b04325dafe2.png">
> 
> It would be nice to have a Select All Button in addition to the Deselect All Button <img alt="Screenshot 2023-06-16 at 2 58 34 PM" width="1625" src="https://user-images.githubusercontent.com/53784701/246522571-ed9c453b-4081-4835-a4b4-bb432178c72e.png"> <img alt="Screenshot 2023-06-16 at 2 58 22 PM" width="1626" src="https://user-images.githubusercontent.com/53784701/246522574-b7bd6da5-b020-4d99-8fdb-343e67908328.png">

**Comment by bronwyncombs** [comment]
Corinna @ CSIRO has expressed their want for this feature again

**Comment by bronwyncombs** [comment]
Silvia @ CSIC has requested this as well

> Specify users at RJB are missing the "select all" button for query results that was available in Specify6.
They told me that, as use case, with a large number of results, selecting all but a few (15 or 25 out of hundreds or thousands) is difficult because they have to select all the records, scrolling down to the last one, and then deselect the few ones.

**Comment by grantfitzsimmons** [comment]
Closing in favor of https://github.com/specify/specify7/issues/7485
